app/server/openshift/src/main/java/io/syndesis/server/openshift/OpenShiftServiceImpl.java

```diff
@@ -310,8 +357,37 @@ private boolean removeSecret(String projectName) {
        return openShiftClient.secrets().withName(projectName).delete();
     }
 
+    private Build waitForBuild(Build r, long timeout, TimeUnit timeUnit) throws InterruptedException {
+        long end = System.currentTimeMillis() + timeUnit.toMillis(timeout);
+        Build next = r;
+
+        while ( System.currentTimeMillis() < end) {
+            if (next.getStatus() != null && ("Complete".equals(next.getStatus().getPhase()) || "Failed".equals(next.getStatus().getPhase()))) {
+                return next;
+            }
+            next = openShiftClient.builds().inNamespace(next.getMetadata().getNamespace()).withName(next.getMetadata().getName()).get();
+            Thread.sleep(5000);
+        }
+        throw SyndesisServerException.launderThrowable(new TimeoutException("Timed out waiting for build completion."));
+    }
+
+    private DeploymentConfig waitForDeployment(DeploymentConfig r, long timeout, TimeUnit timeUnit) throws InterruptedException {
+        long end = System.currentTimeMillis() + timeUnit.toMillis(timeout);
+        DeploymentConfig next = r;
+
+        while ( System.currentTimeMillis() < end) {
+            if (next.getSpec().getReplicas().equals(next.getStatus().getReplicas())) {
+                return next;
+            }
+            next = openShiftClient.deploymentConfigs().inNamespace(next.getMetadata().getNamespace()).withName(next.getMetadata().getName()).get();
+            Thread.sleep(5000);
+        }
+        throw SyndesisServerException.launderThrowable(new TimeoutException("Timed out waiting for build completion."));
+
+    }
+
+
```

Comment:
Would it make sense to use [Awaitility](http://www.awaitility.org/) for the wait* methods?

Reply:
For such a simple task, I am not sure if it worths bringing in an additional dependency (so far its test scoped).

But, I don't really mind.

Reply:
I'm definitely fine as it is. I was thinking if the complexity increases, for instance if we add backoff or we want to schedule it on another thread it would be nice to have this in place. But let's keep it simple for now.